stereo/src/stereo_adaptive_cost_so.cpp

```diff
@@ -36,167 +36,188 @@
  *
  */
 
-/** \brief Adaptive Cost 2-pass Scanline Optimization Stereo Matching algorithm implementation
-  * please see related documentation on stereo/stereo_matching.h
-  *
-  * \author Federico Tombari (federico.tombari@unibo.it)
-  * \ingroup stereo
-  */
+/** \brief Adaptive Cost 2-pass Scanline Optimization Stereo Matching algorithm
+ * implementation please see related documentation on stereo/stereo_matching.h
+ *
+ * \author Federico Tombari (federico.tombari@unibo.it)
+ * \ingroup stereo
+ */
 
 #include "pcl/stereo/stereo_matching.h"
 
 //////////////////////////////////////////////////////////////////////////////
-pcl::AdaptiveCostSOStereoMatching::AdaptiveCostSOStereoMatching ()
+pcl::AdaptiveCostSOStereoMatching::AdaptiveCostSOStereoMatching()
 {
   radius_ = 5;
- 
+
   gamma_s_ = 15;
   gamma_c_ = 25;
- 
+
   smoothness_strong_ = 100;
   smoothness_weak_ = 20;
 }
 
 //////////////////////////////////////////////////////////////////////////////
-void 
-pcl::AdaptiveCostSOStereoMatching::compute_impl (unsigned char* ref_img, unsigned char* trg_img)
+void
+pcl::AdaptiveCostSOStereoMatching::compute_impl(unsigned char* ref_img,
+                                                unsigned char* trg_img)
 {
-  //int n = radius_ * 2 + 1;
-  //int sad_max = std::numeric_limits<int>::max ();
- 
-  float **acc = new float *[width_];
-  for (int d = 0; d < width_; d++)
-  {
+  // int n = radius_ * 2 + 1;
+  // int sad_max = std::numeric_limits<int>::max ();
+
+  float** acc = new float*[width_];
```

Comment:
Can we use some smarter memory management? Even `std::vector<std::vector<float>>` might help. There might be some 2D data-struct in PCL that I don't know about.

Same with the `fwd` and `bck` arrays. The code works, but seems fragile

Reply:
I would absolutely not admit any new code with such memory management. But I'm very hesitant to touch this... I have never used `stereo` module and also I don't know how comprehensive its test coverage is. For me the risks of breaking out-weight the benefits.

Reply:
> 2D data-struct in PCL that I don't know about

Does one exist? What do other modules use? `vector<vector<T>>` is not really a 2D matrix (nor is T**).

> I don't know how comprehensive its test coverage is

Where are the tests? I didn't see a `test/stereo` directory

Reply:
> Does one exist? What do other modules use?

I think `Eigen::MatrixXf` would be an appropriate choice here.

> Where are the tests? I didn't see a `test/stereo` directory. 

Indeed, no test coverage at all. Unfortunately PCL has a number of such dark dusty corners.

> I'll create some tests and refactor this if it's ok?

Any contribution is welcome! That said, if you are willing to volunteer time, I believe there are more important things in PCL that need someone's attention.